dask/bytes/core.py

```diff
@@ -1,22 +1,104 @@
+                lazy=False, **kwargs):
+    """For a list of values which evaluate to byte, produce delayed values
+    which, when executed, result in writing to files.
+
+    The path maybe a concrete directory, in which case it is interpreted
+    as a directory, or a template for numbered output.
+
+    The path may be preceded by a protocol, like ``s3://`` or ``hdfs://`` if
+    those libraries are installed.
+
+    Parameters
+    ----------
+    values: list of Values or dask collection
+        the data to be written
+    urlpath: string
+        Absolute or relative filepaths, URLs (may include protocols like
+        ``s3://``); may be globstring (include `*`).
+    name_function: function or None
+        If using a globstring, this provides the conversion from part number
+        to test to replace `*` with.
+    compression: string or None
+        String like 'gzip' or 'xz'.  Must support efficient random access.
+    **kwargs: dict
+        Extra options that make sense to a particular storage connection, e.g.
+        host, port, username, password, etc.
+
+    Examples
+    --------
+    >>> values = write_bytes(vals, 's3://bucket/part-*.csv')  # doctest: +SKIP
+
+    Returns
+    -------
+    list of ``dask.Delayed`` objects
+    """
+    if hasattr(values, 'to_delayed'):
+        # bag, array or dataframe collection
+        # TODO: better as an isinstance ?
+        values = list(values.to_delayed())
+    if isinstance(urlpath, (tuple, list, set)):
+        storage_options = infer_storage_options(urlpath[0],
+                inherit_storage_options=kwargs)
+        del storage_options['path']
+        paths = [infer_storage_options(u, inherit_storage_options=kwargs)['path']
+                 for u in urlpath]
+    elif isinstance(urlpath, (str, unicode)):
+        storage_options = infer_storage_options(urlpath,
+                                            inherit_storage_options=kwargs)
+        path = storage_options.pop('path')
+        paths = _expand_paths(path, name_function, len(values))
+    else:
+        raise ValueError('URL spec must be string or sequence of strings')
+    if compression == 'infer':
+        if isinstance(urlpath, (tuple, list, set)):
+            compression = infer_compression(urlpath[0])
+        else:
+            compression = infer_compression(urlpath)
+    if compression is not None and compression not in compress_files:
+        raise ValueError("Compression type %s not supported" % compression)
+
+
+    protocol = storage_options.pop('protocol')
+    ensure_protocol(protocol)
+    try:
+        write_bytes = _write_bytes[protocol]
+    except KeyError:
+        raise NotImplementedError("Unknown protocol for writing %s (%s)" %
+                                  (protocol, urlpath))
+
+    out = write_bytes(values, paths, compression=compression, **storage_options)
+    if lazy:
+        return out
+    else:
+        import dask
+        return dask.compute(*out)
```

Comment:
Does this ever get used?  I'd be in favor of removing this option if we can avoid it.  In general I think it's nice to keep the number of options small for core functions if possible.